### PR TITLE
fix the optimizer on some extreme test cases.

### PR DESCRIPTION
**Pull request by wenbingl**

### CODE REVIEW

File: onnxmltools/convert/common/optimizer.py [hunk 1]
```diff
@@ -35,7 +35,8 @@ def is_transpose(self):
 
     @property
     def in_single_path(self):
-        return len(self.successor) <= 1
```
Comment (wschin) [inline onnxmltools/convert/common/optimizer.py]: Do you mind to add a doc string for this function?

File: tests/utils/test_utils.py [hunk 2]
```diff
@@ -106,16 +106,17 @@ def test_optimizer(self):
         nodes[3:] = [helper.make_node('Max', ['input1', 'identity2'], ['max0'])]
         nodes[4:] = [helper.make_node('Transpose', ['max0'], ['tranpose0'], perm=[0, 2, 3, 1])]
         nodes[5:] = [helper.make_node('Transpose', ['tranpose0'], ['tranpose1'], perm=(0, 3, 1, 2))]
+        nodes[6:] = [helper.make_node('Relu', ['tranpose1'], ['output0'], perm=(0, 3, 1, 2))]
```
Comment (wschin) [inline tests/utils/test_utils.py]: This is a very good change. Optimizer should not touch input/output at all.

File: onnxmltools/convert/common/optimizer.py [hunk 3]
```diff
@@ -35,7 +35,8 @@ def is_transpose(self):
...
+        return len(self.successor) == 1 and not self.successor[0].in_or_out and \
+               len(self.precedence) == 1 and len(self.precedence[0].successor) <= 1
 
     @property
     def in_or_out(self):
```
Comment (wschin) [inline onnxmltools/convert/common/optimizer.py]: I don't quite understand this attribute.

Comment (wenbingl) [reply onnxmltools/convert/common/optimizer.py]: Optimization could not happen on fan in/out node.

File: onnxmltools/convert/common/optimizer.py (hunk 1 above)
Comment (wenbingl) [reply onnxmltools/convert/common/optimizer.py]: added

**Pull request merged**